azcopy/run.sh

```diff
@@ -11,13 +16,31 @@ fi
 params+=(--output-level essential)
 params+=(--put-md5)
 
+case $auth_type in
+  sas)
+    bashio::log.info "Using SAS token. Any Service Principal values (client ID, etc) will be ignored";;
+
+  sp)
+    bashio::log.info "Using Service Principal"
+    [[ -z $tenant_id || -z $client_id || -z $client_secret ]] && bashio::exit.nok "tenant_id, client_id and client_secret are required for auth_type=sp"
```

Comment:
@kforeverisback I'm having trouble getting past line 25 in my testing with a SPN. I get the log output I added, but line 25 seems to just silently exit.

```text
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[00:27:24] INFO: Using Service Principal
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped```

Any thoughts?

Reply:
Refactored this logic with bashio functions and all seems to be well (although error handling isn't great).